src/sentry_utils.c

```diff
@@ -409,3 +422,54 @@ sentry__iso8601_to_msec(const char *iso)
 
     return (uint64_t)time * 1000 + msec;
 }
+
+#ifdef SENTRY_PLATFORM_WINDOWS
+#    define sentry__locale_t _locale_t
+#else
+#    define sentry__locale_t locale_t
+#endif
+
+static sentry__locale_t
+sentry__c_locale()
+{
+    static bool c_locale_initialized = false;
+    static sentry__locale_t c_locale;
+    if (!c_locale_initialized) {
+        c_locale_initialized = true;
+#ifdef SENTRY_PLATFORM_WINDOWS
+        c_locale = _create_locale(LC_ALL, "C");
+#else
+        c_locale = newlocale(LC_ALL_MASK, "C", NULL);
+#endif
+    }
+    return c_locale;
+}
+
+double
+sentry__strtod_c(const char *ptr, char **endptr)
+{
+#ifdef SENTRY_PLATFORM_WINDOWS
+    return _strtod_l(ptr, endptr, sentry__c_locale());
+#else
+    return strtod_l(ptr, endptr, sentry__c_locale());
+#endif
+}
+
+int
+sentry__snprintf_c(char *buf, size_t buf_size, const char *fmt, ...)
+{
+    va_list args;
+    va_start(args, fmt);
+
+    int rv;
+#ifdef SENTRY_PLATFORM_WINDOWS
+    rv = _vsnprintf_l(buf, buf_size, fmt, sentry__c_locale(), args);
+#elif defined(SENTRY_PLATFORM_MACOS)
+    rv = vsnprintf_l(buf, buf_size, sentry__c_locale(), fmt, args);
+#else
+    rv = stbsp_vsnprintf(buf, buf_size, fmt, args);
```

Comment:
So I think this is fine, but in theory we could alternatively make a branch here that uses `uselocale` to switch the locale for the current thread and restore it. In the grand scheme of stupid things, that might be an alternative.

Reply:
not sure I want to mess with that, I kept it as is for now

Reply:
I think it's fine keeping it like this. We could evaluate it later if we want to trim down on bloat.